crates/proto/src/h2/h2_client_stream.rs

```diff
@@ -466,7 +475,10 @@ where
                     match ServerName::try_from(&*tls.dns_name) {
                         Ok(dns_name) => {
                             let tls = TlsConnector::from(tls.client_config);
-                            let tls = tls.connect(dns_name.to_owned(), AsyncIoStdAsTokio(tcp));
+                            let tls = Box::pin(time::timeout(
```

Comment:
Nit: I'd prefer to use `timeout()` directly, I don't think the `time::` qualification adds anything other than clutter.